ruby/hashes.rb

```diff
@@ -0,0 +1,118 @@
+# Interior Designer Client Form
+
+# enter client name
+puts "Please enter your name:"
+client_name = gets.chomp
+
+# enter client age
+puts "Please enter your age:"
+client_age = gets.chomp
+
+# enter client's number of children, make until loop to ensure correct data type
+puts "Please enter your number of children (digits only):"
+number_children = gets.chomp
+
+# enter client's decor theme
+puts "Please enter your decor theme:"
+decor_theme = gets.chomp
+
+# enter client's favorite color
+puts "Please enter your favorite color:"
+favorite_color = gets.chomp
+
+# enter whether client prefers the regular package or the deluxe package
+puts "Do you prefer the regular suite or the deluxe suite? (please enter regular/deluxe)"
+suite_pref = gets.chomp
+
+# keep prompting user until they enter either "regular" or "deluxe"
+until suite_pref == "regular" || suite_pref == "deluxe"
+  puts "Incorrect input."
+  puts "Do you prefer the regular suite or the deluxe suite? (please enter regular/deluxe)"
+  suite_pref = gets.chomp
+end
+
+# set deluxe equal to true or false based on user input
+if suite_pref == "deluxe"
+  deluxe = true
+else
+  deluxe = false
+end
+
+# put user input into hash
+decor_form = {
+  name: client_name,
+  age: client_age.to_i,
+  children: number_children.to_i,
+  theme: decor_theme,
+  color: favorite_color,
+  suite: deluxe
+}
+
+# print hash
+p decor_form
+
+
+# ask client if they want to update a key
+puts "Do you want to update a key? Enter key you'd like to update, or enter 'none'."
+update = gets.chomp
+
+# make sure client input corresponds to key name or is 'none'... handling user errors just for fun
+until update == "none" || update == "name" || update == "age" || update == "children" || update == "theme" || update == "color" || update == "suite"
+  puts "I didn't understand you. Please enter key name or type 'none'."
+  update = gets.chomp
+end
+
+# if the user does not want to change any input, do nothing
+# else if the user wants to change any of the key values,
+# prompt them to type in the category they want to change.
+# once they do this, ask them what they want to set the new value to,
+# then substitute the value in the hash
+if update == "none"
+  nil
```

Comment:
converting a string to a symbol is a practice in finding a dynamic way to update your values by accessing the key.

Think about what data type the client keys are and data type gets.chomp returns and how much cleaner and DRY(Don't Repeat Yourself)'r your code would be if you had access to the keys.